### ISSUE TITLE
Bug Report: Plug-in lost sync with Xfinity with no errors. Now crashes with 400 error on startup

### ISSUE THREAD

**Issue by rdeutsch3** [bloomkd46/homebridge-XfinityHome#39]
### Describe The Bug:

Initially the plug in seemed to loose communication with the Xfinity service as it would not control the alarm.  I restarted the child bridge and now the process crashes with a 400 error.  Subsequently the child bridge continually restarts.

Note: the plugin was working fine on 15 DEC and seemed to go awry sometime on the 15th or 16th and has not been operational since.  Nothing was changed on the Pi, plug-in, etc. before it failed.

### To Reproduce:

Restart child bridge. 

### Expected behavior:

Normal communication with the Xfinity service

### Logs:

```shell
[18/12/2023, 10:58:16] [Xfinity Home System] Restarting Process...
[18/12/2023, 10:58:17] [Xfinity Home System] Launched child bridge with PID 27251
[18/12/2023, 10:58:18] Registering platform 'homebridge-xfinityhome.XfinityHomePlatform'
[18/12/2023, 10:58:18] [Xfinity Home System] Loaded homebridge-xfinityhome v3.4.1 child bridge successfully
[18/12/2023, 10:58:18] Loaded 11 cached accessories from cachedAccessories.0E5DDF42C235.
[18/12/2023, 10:58:18] [Xfinity Home System] Finished initializing platform: Xfinity Home System
[18/12/2023, 10:58:18] [Xfinity Home System] Loading accessory from cache: Panel
[18/12/2023, 10:58:18] [Xfinity Home System] Loading accessory from cache: Entry Doors
[18/12/2023, 10:58:18] [Xfinity Home System] Loading accessory from cache: Master Glass Window Break
[18/12/2023, 10:58:18] [Xfinity Home System] Loading accessory from cache: Family Glass Window Break
[18/12/2023, 10:58:18] [Xfinity Home System] Loading accessory from cache: Breakfast Windows
[18/12/2023, 10:58:18] [Xfinity Home System] Loading accessory from cache: Front Windows
[18/12/2023, 10:58:18] [Xfinity Home System] Loading accessory from cache: 2nd Floor Windows
[18/12/2023, 10:58:18] [Xfinity Home System] Loading accessory from cache: Master Windows
[18/12/2023, 10:58:18] [Xfinity Home System] Loading accessory from cache: Living Windows
[18/12/2023, 10:58:18] [Xfinity Home System] Loading accessory from cache: Downstairs
[18/12/2023, 10:58:18] [Xfinity Home System] Loading accessory from cache: Upstairs
[18/12/2023, 10:58:18] Publishing bridge accessory (name: Xfinity Home System, publishInfo: {
  username: '0E:5D:DF:42:C2:35',
  port: 32064,
  pincode: '***-**-***',
  category: 2,
  bind: [ [length]: 0 ],
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: 'ciao'
}).
[18/12/2023, 10:58:18] [Xfinity Home System] Executed didFinishLaunching callback
[18/12/2023, 10:58:18] [Xfinity Home System] Using Refresh Token From Config: zzzzz
[18/12/2023, 10:58:18] [Xfinity Home System] Loaded 11 Accessories From Cache
[18/12/2023, 10:58:18] Homebridge v1.7.0 (HAP v0.11.1) (Xfinity Home System) is running on port 32064.

/var/lib/homebridge/node_modules/homebridge-xfinityhome/node_modules/axios/lib/core/createError.js:16
  var error = new Error(message);
              ^
Error: Request failed with status code 400
    at createError (/var/lib/homebridge/node_modules/homebridge-xfinityhome/node_modules/axios/lib/core/createError.js:16:15)
    at settle (/var/lib/homebridge/node_modules/homebridge-xfinityhome/node_modules/axios/lib/core/settle.js:17:12)
    at IncomingMessage.handleStreamEnd (/var/lib/homebridge/node_modules/homebridge-xfinityhome/node_modules/axios/lib/adapters/http.js:322:11)
    at IncomingMessage.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1408:12)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
[18/12/2023, 10:58:19] [Xfinity Home System] Child bridge process ended
[18/12/2023, 10:58:19] [Xfinity Home System] Process Ended. Code: 1, Signal: null
[18/12/2023, 10:58:25] [Xfinity Home System] Stopping child bridge (will not restart)...
```

### Config:

```shell
{
    "bridge": {
        "name": "XXXX",
        "username": "YYYY",
        "port": 51035,
        "pin": "XXX-XX-XXX",
        "advertiser": "ciao",
        "bind": []
    },
    "accessories": [],
    "platforms": [
        {
            "name": "UI Config",
            "port": 8581,
            "auth": "form",
            "theme": "auto",
            "tempUnits": "f",
            "lang": "auto",
            "platform": "config"
        },
        {
            "name": "Xfinity Home System",
            "pin": "1597",
            "refreshToken": "XXXX",
            "temperatureSensors": true,
            "logLevel": 3,
            "hideUnsupportedDeviceWarnings": false,
            "logWatchdogErrors": false,
            "_bridge": {
                "username": "XXXXXX",
                "port": 32064
            },
            "platform": "XfinityHomePlatform"
        }
    ]
}
```

### Operating System:

Raspberian

### Node Version:

v20.9.0 

### NPM Version:

v10.1.0

### Homebridge Version:

v1.7.0

### Plugin Version:

v3.4.1

**Comment by rdeutsch3** [comment]
Although this could be the same issue as #38, I've logged it separately as my system was working fine for quite some time and just stopped working.

**Comment by bloomkd46** [comment]
Does the Xfinity Home app still work on your phone? I had this error come up two weeks ago and it was a result of Xfinity resetting my password. (This was on my alt account). Another week later my primary account needed the password reset.

Let me know if that doesn't solve the problem

**Comment by rdeutsch3** [comment]
Bingo!  I refreshed the token and all good again.

BTW: I logged #27 which asks for some updates to the proxy instructions.  I've subsequently abandoned use of my Synology, for eventual obsolescence Homebridge reasons, but there is one point that is still very germane:

Add a Note something like the following:
-Ensure the default iOS browser is set to Safari, for the proxy certificate to be installed into the iOS Settings.

Rationale: Only Safari worked for me. Firefox just kept downloading the certificate, but not allowing me to do anything with it.

I scratched my head a bit, then I recalled something about Safari, and all good.  I then recalled I logged an enhancement to update the instructions.  The bullet point above can help others.

FYI: I was able to get the proxy tool to work and I'm running Node.js v20.9.0, so this may help you close #37.  :)

Thank you sir.

**Comment by home-automate** [comment]
Hi, 

Just adding here I too was able to resolve the same issue (running node 20.10.0 / npm 10.2.3)

I had the same 400 error, though just changing the pw wasn’t enough. Below the sequence I used:
1. disable xfinity home homebridge plugin
2. Update password on Xfinity site
3. From the settings of the plugin in HomeBridge, setup the proxy
4. Login to the xfinity app
5. Confirm token has been captured
6. Uninstall xfinity app
7. Enable xfinity home homebridge plugin
8. Restart homebridge services
9. Everything working again

**Comment by rdeutsch3** [comment]
> Hi,
> 
> Just adding here I too was able to resolve the same issue (running node 20.10.0 / npm 10.2.3)
> 
> I had the same 400 error, though just changing the pw wasn’t enough. Below the sequence I used:
> 
> 1. disable xfinity home homebridge plugin
> 2. Update password on Xfinity site
> 3. From the settings of the plugin in HomeBridge, setup the proxy
> 4. Login to the xfinity app
> 5. Confirm token has been captured
> 6. Uninstall xfinity app
> 7. Enable xfinity home homebridge plugin
> 8. Restart homebridge services
> 9. Everything working again

Your response is clearer than mine.  This is exactly what I did as well.  However, I had to set my default browser to Safari in order for the certificate to be installed, hence my requested update to the instructions.

**Comment by home-automate** [comment]
Thanks for the Safari call out. I guess there are two enhancement requests:
1. Update the instructions in the proxy setup to ensure that Safari is used to retrieve the certificate
2. Have exception handling for the 400 error to allow the startup of homebridge to continue and indicate that user has to verify the login/update the token

**Comment by rdeutsch3** [comment]
> Thanks for the Safari call out. I guess there are two enhancement requests:
> 
>     1. Update the instructions in the proxy setup to ensure that Safari is used to retrieve the certificate
> 
>     2. Have exception handling for the 400 error to allow the startup of homebridge to continue and indicate that user has to verify the login/update the token

True.  I'm not sure how or when @bloomkd46 might like to address "the Safari" note in the instructions as it is also called out in #27.  The exception and message about invalid oauth token make things clearer vs. crashing on the 400 error.  The interesting part of the token expiration is there was nothing in the logs to indicate anything was wrong.  I only got the error messages when I restarted the child bridge.

**Comment by bloomkd46** [comment]
I did just get notified by Xfinity the reason for the password reset. [here](https://assets.xfinity.com/assets/dotcom/learn/Data-Incident.pdf) is the link for those of you who didn’t get the email

**Comment by rdeutsch3** [comment]
> I did just get notified by Xfinity the reason for the password reset. [here](https://assets.xfinity.com/assets/dotcom/learn/Data-Incident.pdf) is the link for those of you who didn’t get the email

Thank you.  I have not received the email, but saw it on the national news this am.

**Comment by bloomkd46** [comment]
Fixed v3.4.2
